libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/HttpRequest.cs

```diff
@@ -228,187 +228,196 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
+                    }
                 }
-            }
 
-            // if there no usr-agent in the header, set the user-agent to Mozzila/5.0
-            if (!client.DefaultRequestHeaders.Contains("user-agent"))
-            {
-                client.DefaultRequestHeaders.TryAddWithoutValidation("user-agent", "Mozilla/5.0");
-            }
+                request.Headers.TryAddWithoutValidation("user-agent", "Mozilla/5.0");
 
-            dynamic traceInfo = new JObject();
+                dynamic traceInfo = new JObject();
 
-            traceInfo.request = new JObject();
-            traceInfo.request.method = this.Method.ToString();
-            traceInfo.request.url = instanceUrl;
+                traceInfo.request = new JObject();
+                traceInfo.request.method = this.Method.ToString();
+                traceInfo.request.url = instanceUrl;
 
-            HttpResponseMessage response = null;
-            string contentType = ContentType?.GetValue(dc.State) ?? "application/json";
+                HttpResponseMessage response = null;
+                string contentType = ContentType?.GetValue(dc.State) ?? "application/json";
 
-            switch (this.Method)
-            {
-                case HttpMethod.POST:
-                    if (instanceBody == null)
-                    {
-                        response = await client.PostAsync(instanceUrl, content: null, cancellationToken: cancellationToken).ConfigureAwait(false);
-                    }
-                    else
-                    {
-                        using (var postContent = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType))
+                switch (this.Method)
+                {
+                    case HttpMethod.POST:
+                        if (instanceBody == null)
                         {
-                            traceInfo.request.content = instanceBody.ToString();
-                            traceInfo.request.headers = JObject.FromObject(postContent?.Headers.ToDictionary(t => t.Key, t => (object)t.Value?.FirstOrDefault()));
-                            response = await client.PostAsync(instanceUrl, postContent, cancellationToken).ConfigureAwait(false);
+                            response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                        }
+                        else
+                        {
+                            using (var postContent = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType))
+                            {
+                                traceInfo.request.content = instanceBody.ToString();
+                                traceInfo.request.headers = JObject.FromObject(postContent?.Headers.ToDictionary(t => t.Key, t => (object)t.Value?.FirstOrDefault()));
+                                request.Content = new StringContent(instanceBody.ToString(), Encoding.UTF8, contentType);
```

Comment:
Isn't this the postContent we created in the using in line 299? Shouldn't we be using that disposable instance instead of creating a new instance of StringContent?